### ISSUE TITLE
Determine Isolate default for Cycle-UI Components

### ISSUE THREAD

**Issue by arlair** [eldarlabs/cycle-ui#3]
See https://github.com/cyclejs/core/issues/259 for a large discussion.

**Current thinking**
- If I create a Cycle Component, I want a sane default, but without [Cycle Isolate](https://github.com/cyclejs/isolate) you don't really know if the Intent implementation of the Cycle Component will modify other instances when additional Cycle Components are created.
- The only way to determine if the Cycle Component has cascading intents (like CSS, but the opposite direction?) is to check the source code of the component or try to test the behaviour at runtime. This feels like every Cycle Component is possibly a loaded gun and it is the responsibility of the client to read the source to determine if it is going to start influencing other controls. If I ask for a Slider component, I would expect to get one not linked to other slider components by default as that is how all UI libraries I recall working with have worked.
- I don't understand the reason why the article below makes isolation not desirable.

> Another way to phrase this is: when is isolation not desirable?
> There is one case I know of: http://staltz.com/adapting-controlled-and-uncontrolled-fields.html  check theInput component
- I'm possibly missing something fundamental here about Cycle Components due to my non functional programming background or own stupidity ;).

**Comment by wclr** [comment]
Cycle is very explicit and that is one of its core pros. Calling `isolate` explicity also increases expressiveness wich is good, but one can always create isolated "by default" version of a component.

**Comment by theefer** [comment]
> Calling isolate explicity also increases expressiveness wich is good, but one can always create isolated "by default" version of a component.

This implies that some components are de-facto isolated, while others aren't. The only way to check whether `isolate` needs adding is once again to read the source code. Having to superstitiously review a component's implementation doesn't bode well for sharing libraries of reusable components e.g. via npm.

**Comment by arlair** [comment]
> This implies that some components are de-facto isolated, while others aren't

I think this may be true. Something like the [App component](https://github.com/eldarlabs/cycle-ui/blob/master/src/app/App.ts) is just a div wrapper with a  class, it doesn't modify other components. I don't think it needs isolation. It is more of an optimisation because isolation would not stop it working, but would add some unnecessary classes to the HTML output. But perhaps defensive programming is safer?

I think by default isolation responsibility should live within the component and not be leaked outside for all consumers to worry about. The component should play well with other instances of itself and other similar components. It should not require easily forgotten calls to tame it everywhere it is used, especially as these can introduce subtle bugs only visible at runtime.

**Comment by wclr** [comment]
Cycle approach is very explicit an consice, so must me a user that uses it. If you don't want/need this explity/conscience/logical verbosity you don't want to use cycle: there are many alternatives that "hide" rough spots from user and do "magical" stuff. imo.

**Comment by theefer** [comment]
> I think by default isolation responsibility should live within the component and not be leaked outside for all consumers to worry about. The component should play well with other instances of itself and other similar components. It should not require easily forgotten calls to tame it everywhere it is used, especially as these can introduce subtle bugs only visible at runtime.

I agree. This is what's being discussed at https://github.com/cyclejs/core/issues/259 to see if it can be resolved at the framework level, rather than in each library or app using Cycle.

**Comment by arlair** [comment]
@TylorS patiently walked me through to some understanding for the current isolation mechanism in Cycle.js. I'm not sure the current isolation implementation allows me to easily achieve my goals with this library, but as mentioned here, it may change in the future. https://github.com/TylorS/cycle-snabbdom/issues/29#issuecomment-213075925

Other alternatives would be to try [TSERS](https://github.com/tsers-js/core) where isolation seems to happen implicitly because of the DOM preparation and the true MVI vs Cycle's IMV. Although it sounds like that [may have some changes too](https://gitter.im/tsers-js/chat?at=570be963c65c9a6f7f28abec).
